### PR TITLE
Created project Code of Conduct

### PR DESCRIPTION
**Pull request by crcrewso**
# Pull Request Template

## Summary

This project needs community guidelines as the project goes forward trying to increase user and developer engagement, including rules for AI use. 

## Related Issues

Satisfies part of #690 

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature ( adds functionality, always assumed to be breaking until tested by maintainers)
- [ ] Breaking change (fix or addition that changes existing behaviour)
- [x] Documentation only

## Target Branch

develop

## AI Usage Disclosure

- [ ] No AI tools were used in this PR.
- [x] AI tools were used:
  - **Tools / where used:** _e.g. GitHub Copilot for boilerplate_
  - [x] I've reviewed every AI-generated line as if I wrote it myself, and confirm: no PHI shared, no license-incompatible content (Apache-2.0).

The first draft of these documents were made by asking AI to consider Django and python codes of conduct, as well as my (@crcrewso) language and tone. 

---

### Coder Tasks

- [x] Rebased / merged `develop` branch; no merge conflicts.

All other normal coding tasks don't apply as the only changes are root directory markdown files that are only visible in the code base and on GitHub
- [x] `uv run pre-commit run --all-files` passes (ruff lint, ruff-format, django-upgrade).
  - `uv run ruff check . --fix` to help resolve linting errors
- [x] `uv run python -Wd manage.py check` reviewed for new deprecation warnings.
- [ ] `uv run pytest` full test suite passes.
- [ ] `uv run python manage.py makemigrations --check --dry-run` if migrations are found, change should be marked Breaking
- [ ] Docker builds succeed if touched.
- [ ] Docs updated under `docs/` if behaviour, settings, or APIs changed.
- [ ] `release_notes.md` updated if user-facing.
- [ ] No secrets, debug code, or stray `print`/`console.log` left in.

### Review Code

- [x] Reviewer verifies the target branch.
- [x] For non-trivial changes: pulled the branch, ran it locally, and confirmed it works as described.
- [x] CI is green; test suite and `pre-commit` pass.
- [x] Reads through all changed files for anything weird or unintended.
- [x] Logic is correct; edge cases and error handling considered.
- [x] Migrations are sane and reversible where possible.
- [x] Changes are consistent with the modernization direction (no reintroduction
      of patterns being migrated away from, e.g. flake8/yapf-era conventions).
- [x] No security concerns (input validation, permissions, injection, XSS).
- [x] Comments are sufficient and not overkill.
- [x] Typos, spellcheck, professional.
- [x] Reviewer sends back review.

### Edits

- [ ] Coder fixes any issues.
- [ ] Reviewer approves.

> Maintainer review is required for breaking changes, new features, and migrations.
> For bug fixes, refactors, and docs-only changes, reviewer approval is sufficient to merge.

---

## Testing Notes

This commit only changes repository level markdown changes. No changes have been made to other documentation

## Screenshots

N/A

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+```bash
+python manage.py test
+# or use the helper script:
+bash runtests.sh
+```
+
+### Opening a pull request
+
+1. Push your branch to your fork.
+2. Open a PR against the `Dev` branch (not `master`).
```
Comment (nsmela) [inline CONTRIBUTING.md]: 'develop' branch now

File: AGENTS.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+(GitHub Copilot, Claude Code, Cursor, etc.) and to fully automated AI agents
+exploring or drafting changes in this repository.
+
+---
+
+## Project overview
+
+| | |
+|---|---|
+| **Language** | Python 3.12 / 3.13 |
```
Comment (nsmela) [inline AGENTS.md]: only 3.12

File: AGENTS.md [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+| **Framework** | Django 4.2 (LTS) |
+| **Database** | PostgreSQL, MySQL, or MS SQL Server |
```
Comment (nsmela) [inline AGENTS.md]: pls note MySQL is legacy support

File: AGENTS.md [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+| **Package manager** | [uv](https://docs.astral.sh/uv/) (preferred) or pip |
+| **Linter / formatter** | [ruff](https://docs.astral.sh/ruff/) |
+| **Test runner** | pytest via `runtests.sh` or `python manage.py test` |
+| **Docs** | Sphinx — `make docs` from repo root |
+| **Target branch** | `Dev` (not `master`) |
```
Comment (nsmela) [inline AGENTS.md]: target branch is 'develop'

File: AGENTS.md [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+
```
Comment (nsmela) [inline AGENTS.md:75]: should we note `uv` is also used?

File: AGENTS.md [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+## Repository layout
+
+```
+qatrack/        # Django project root; most application code lives here
+  accounts/     # User accounts and authentication
+  api/          # Django REST Framework API
+  attachments/  # File-attachment handling
+  contacts/     # Contact / notification system
+  core/         # Core utilities, mixins, templatetags
```
Comment (nsmela) [inline AGENTS.md]: should be `qatrack_core`

File: AGENTS.md [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+| **Test runner** | pytest via `runtests.sh` or `python manage.py test` |
+| **Docs** | Sphinx — `make docs` from repo root |
+| **Target branch** | `Dev` (not `master`) |
+
+---
+
+## Repository layout
+
+```
+qatrack/        # Django project root; most application code lives here
```
Comment (nsmela) [inline AGENTS.md]: no mention of accel_migration_tool

File: AGENTS.md [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+  accounts/     # User accounts and authentication
```
Comment (nsmela) [inline AGENTS.md]: missing admin_media

File: AGENTS.md [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+  api/          # Django REST Framework API
+  attachments/  # File-attachment handling
+  contacts/     # Contact / notification system
```
Comment (nsmela) [inline AGENTS.md]: missing cache

File: AGENTS.md [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+| **Database** | PostgreSQL, MySQL, or MS SQL Server |
+| **Package manager** | [uv](https://docs.astral.sh/uv/) (preferred) or pip |
+| **Linter / formatter** | [ruff](https://docs.astral.sh/ruff/) |
+| **Test runner** | pytest via `runtests.sh` or `python manage.py test` |
+| **Docs** | Sphinx — `make docs` from repo root |
+| **Target branch** | `Dev` (not `master`) |
+
+---
+
+## Repository layout
```
Comment (nsmela) [inline AGENTS.md:78]: whole layout is missing quite a few folders.

File: AGENTS.md [hunk 11]
```diff
@@ -0,0 +1,275 @@
...
+PR or call them out explicitly in the PR description under a
+**"📚 Documentation to review"** heading.
+
+A GitHub Actions workflow (`.github/workflows/docs-impact.yml`) posts an
+automated heuristic comment on every PR that lists which doc paths may need
+attention. The mapping it uses is reproduced here so that AI agents and
+developers can apply the same logic before a PR is even opened.
+
+The repository also includes an annual workflow
+(`.github/workflows/annual-repository-health-review.yml`) that creates or
```
Comment (nsmela) [inline AGENTS.md]: file does not exist

File: AGENTS.md [hunk 12]
```diff
@@ -0,0 +1,275 @@
...
+---
+
+## Documentation impact
+
+When making code changes, check whether any of the following documentation
+pages may be affected. If they are, either update the docs as part of the same
+PR or call them out explicitly in the PR description under a
+**"📚 Documentation to review"** heading.
+
+A GitHub Actions workflow (`.github/workflows/docs-impact.yml`) posts an
```
Comment (nsmela) [inline AGENTS.md]: file does not exist

File: AGENTS.md [hunk 13]
```diff
@@ -0,0 +1,275 @@
...
+  faults/       # Fault-logging application
+  notifications/
+  parts/        # Spare-parts tracking
+  qa/           # QA test-list engine — the heart of the application
+  reports/
+  service_log/  # Service and maintenance logs
+  units/        # Treatment-unit definitions
+docs/           # Sphinx documentation (reStructuredText)
+fixtures/       # Demo / seed data
+requirements/   # Pinned requirement files (base, dev, docs)
```
Comment (nsmela) [inline AGENTS.md]: description is inaccurate: only dev.txt remains

File: AGENTS.md [hunk 14]
```diff
@@ -0,0 +1,275 @@
...
+## Project overview
+
+| | |
+|---|---|
+| **Language** | Python 3.12 / 3.13 |
+| **Framework** | Django 4.2 (LTS) |
+| **Database** | PostgreSQL, MySQL, or MS SQL Server |
+| **Package manager** | [uv](https://docs.astral.sh/uv/) (preferred) or pip |
+| **Linter / formatter** | [ruff](https://docs.astral.sh/ruff/) |
+| **Test runner** | pytest via `runtests.sh` or `python manage.py test` |
```
Comment (nsmela) [inline AGENTS.md]: runtests.sh does not use pytest. should we consider removing the script entirely? is it being used?

File: CODE_OF_CONDUCT.md [hunk 15]
```diff
@@ -0,0 +1,67 @@
+# Code of Conduct
+
+## Our Pledge
```
Comment (nsmela) [inline CODE_OF_CONDUCT.md:3]: The protected-characteristics list omits sexual orientation. The pledge covers "age, background, disability, ethnicity, level of experience, nationality, personal appearance, race, religion, or gender identity" — but sexual orientation is missing, along with gender expression and sex characteristics. This is conspicuous precisely because the Attribution section says the document is informed by Contributor Covenant v2.1, whose list explicitly includes sexual orientation. For a healthcare/academic community CoC, dropping it is the kind of omission people notice and read meaning into. I'd add at least sexual orientation and gender identity and expression. ("background" is also vague and somewhat redundant next to ethnicity/nationality — optional to drop.)

File: CODE_OF_CONDUCT.md [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+
+**[medphys@crcrewso.ca](mailto:medphys@crcrewso.ca)**
+
+Reports are handled confidentially. The maintainer will review the situation
+and respond as promptly as possible. Consequences may range from a private
+reminder to a temporary or permanent ban, depending on the severity.
+
+If your report concerns the maintainer directly, please reach out to another
+long-standing contributor listed in
+[CONTRIBUTORS.txt](CONTRIBUTORS.txt).
```
Comment (nsmela) [inline CODE_OF_CONDUCT.md]: has no contact info for contributors, no indication if they're active, etc. not sure if showing email addresses is a good option, maybe the @nsmela approach?

File: AGENTS.md
Comment (nsmela) [inline AGENTS.md:1]: we should revisit it and see if we can condense it a bit further

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -0,0 +1,170 @@
...
+   ```
+5. Start the development server:
+   ```bash
+   python manage.py runserver
+   ```
+
+### Coding guidelines
+
+- Follow existing code style. The project uses
+  [flake8](https://flake8.pycqa.org/) for linting and
```
Comment (nsmela) [inline CONTRIBUTING.md]: we use `ruff` now

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -0,0 +1,170 @@
...
+  [isort](https://pycqa.github.io/isort/) for import ordering.
```
Comment (nsmela) [inline CONTRIBUTING.md]: using `ruff`

File: CONTRIBUTING.md [hunk 19]
```diff
@@ -0,0 +1,170 @@
...
+  unnecessary jargon.
+
+### Running the tests
+
+```bash
+python manage.py test
+# or use the helper script:
+bash runtests.sh
+```
+
```
Comment (nsmela) [inline CONTRIBUTING.md:153]: need to mention `uv run pre-commit --all-files`

File: CONTRIBUTING.md [hunk 20]
```diff
@@ -0,0 +1,170 @@
...
+   ```
+4. Apply migrations and load fixture data:
+   ```bash
+   python manage.py migrate
+   python manage.py loaddata fixtures/demo_data.json
+   ```
+5. Start the development server:
+   ```bash
+   python manage.py runserver
+   ```
```
Comment (nsmela) [inline CONTRIBUTING.md:112]: pre-commit install should be used

File: CONTRIBUTING.md [hunk 21]
```diff
@@ -0,0 +1,170 @@
...
+
+- **Report bugs** or **request features** by opening an issue.
+- **Improve the documentation** in the `docs/` folder (see below).
+- **Write a tutorial** showing how to use a QATrack+ feature.
+- **Translate** the interface into another language — QATrack+ marks its strings
+  for translation, and localization is an active area of work.
+- **Answer questions** and share your experience in
+  [GitHub Discussions](https://github.com/qatrackplus/qatrackplus/discussions).
+- **Share known issues and solutions** in the
+  [Common Issues & Solutions](docs/common_issues/) documentation section.
```
Comment (nsmela) [inline CONTRIBUTING.md]: directory does not exist

File: CONTRIBUTING.md [hunk 22]
```diff
@@ -0,0 +1,170 @@
...
+the [QATrack+ Google Group](https://groups.google.com/g/qatrack). The
+[project wiki](https://github.com/qatrackplus/qatrackplus/wiki) collects FAQs and
+community guides. For anything not covered by those channels, email
+[medphys@crcrewso.ca](mailto:medphys@crcrewso.ca).
+
+## Reporting bugs
+
+1. Check the [existing issues](https://github.com/qatrackplus/qatrackplus/issues)
+   first — your bug may already be reported.
+2. Open a new issue and fill in the bug report template. Include:
```
Comment (nsmela) [inline CONTRIBUTING.md]: we do not have a bug report template

File: CONTRIBUTING.md [hunk 23]
```diff
@@ -0,0 +1,170 @@
...
+
+The documentation lives in `docs/` and is built with
+[Sphinx](https://www.sphinx-doc.org). Most pages use
+[reStructuredText](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html)
+(`.rst`).
+
+To build the docs locally:
+
+```bash
+pip install -r requirements/docs.txt   # or: pip install sphinx
```
Comment (nsmela) [inline CONTRIBUTING.md]: file does not exist

File: CONTRIBUTING.md (hunk 1 above)
Comment (nsmela) [inline CONTRIBUTING.md]: `develop` branch

**Review by nsmela — CHANGES_REQUESTED** [review]

**Comment by crcrewso** [comment]
I should have mentioned AGENTS.md is a human readable file intended mostly for AI agents to us. There are quite a few comments I should have caught, especially in the agents file. Thank you

File: CONTRIBUTING.md (hunk 1 above)
Comment (crcrewso) [reply CONTRIBUTING.md]: addressed in 7e594ad2e9ce341c01f614ecf2958fa08e601015

File: AGENTS.md (hunk 3 above)
Comment (crcrewso) [reply AGENTS.md]: addressed in 7e594ad2e9ce341c01f614ecf2958fa08e601015 
I believe legacy will be the appropriate term after we have declared official deprecation (4.1?)

File: AGENTS.md (hunk 5 above)
Comment (crcrewso) [reply AGENTS.md:75]: addressed in 7e594ad2e9ce341c01f614ecf2958fa08e601015

File: AGENTS.md (hunk 9 above)
Comment (crcrewso) [reply AGENTS.md]: I had erroneously thought cache was generated with the createcachetable command. I now know better.

File: AGENTS.md (hunk 11 above)
Comment (crcrewso) [reply AGENTS.md]: Great catch! This is a part of #801 that needs to be split out.

File: AGENTS.md (hunk 13 above)
Comment (crcrewso) [reply AGENTS.md]: Future PR to close the loop, including new instructions on how to use pip with the pyproject file directly.

File: AGENTS.md (hunk 14 above)
Comment (crcrewso) [reply AGENTS.md]: probably, but I feel reviewing this file is out of scope of this PR.

File: CODE_OF_CONDUCT.md (hunk 15 above)
Comment (crcrewso) [reply CODE_OF_CONDUCT.md:3]: I had a bit of an internal debate when I wrote this section. You are 100% right, genuinely, thank you for holding me to this one!

File: CODE_OF_CONDUCT.md (hunk 16 above)
Comment (crcrewso) [reply CODE_OF_CONDUCT.md]: addressed in 7e594ad2e9ce341c01f614ecf2958fa08e601015

File: AGENTS.md
Comment (crcrewso) [reply AGENTS.md:1]: The updated version of this file correctly expresses the document's intent, what parts should be read by all contributors, and reorders them appropriately. I believe the intent of the AGENTS.md convention is mostly for agents, if this changes this document will need to be split.

File: CONTRIBUTING.md (hunk 22 above)
Comment (crcrewso) [reply CONTRIBUTING.md]: Should we? It's something to revisit in the near future .

File: AGENTS.md [hunk 24]
```diff
@@ -0,0 +1,319 @@
...
+---
+
+## Getting started
+
+```bash
+# 1. Create a virtual environment and install all dependencies
+uv sync --dev
+
+# 2. Provide local settings (database credentials, etc.)
+cp qatrack/local_settings.example.py qatrack/local_settings.py
```
Comment (nsmela) [inline AGENTS.md:135]: [qatrack/local_settings.example.py](https://github.com/qatrackplus/qatrackplus/blob/develop/qatrack/local_settings.example.py) does not exist. Referenced in AGENTS.md Getting started step 2 and CONTRIBUTING.md step 4. The real path, per [docs/developer/guide.rst:165-171](https://github.com/qatrackplus/qatrackplus/blob/develop/docs/developer/guide.rst#L165-L171), is:

```
cp deploy/dev/local_settings.dev.py qatrack/local_settings.py
cp deploy/dev/local_test_settings.dev.py qatrack/local_test_settings.py
```

File: AGENTS.md [hunk 25]
```diff
@@ -0,0 +1,319 @@
...
+# edit qatrack/local_settings.py as needed
+
+# 3. Apply migrations and load demo data
+uv run python manage.py migrate
+uv run python manage.py loaddata fixtures/demo_data.json
```
Comment (nsmela) [inline AGENTS.md:140]: `fixtures/demo_data.json` does not exist (AGENTS.md step 3). `fixtures/` contains only `defaults/{qa,service_log,units}`. `CONTRIBUTING.md` already has the correct `loaddata fixtures/defaults/*/*.json` — so the two new files contradict each other on the very first setup run.

File: AGENTS.md [hunk 26]
```diff
@@ -0,0 +1,319 @@
...
+> agent tool-call environments start a fresh shell for every command, so an
+> activation from a previous step won't carry over — `uv run` re-resolves
+> the correct environment each time regardless. Human contributors are of
+> course welcome to `source .venv/bin/activate` once and drop the prefix.
+
+---
+
+## Linting and formatting
+
+```bash
```
Comment (nsmela) [inline AGENTS.md:157]: need to note that repo-wide formatting hasn't been applied yet. a ruff check might catch tonnes to reformat. new contributors should scope to the files they changed

Comment (nsmela) [reply AGENTS.md:157]: we should only encourage doing a ruff check, not the format

File: AGENTS.md [hunk 27]
```diff
@@ -0,0 +1,319 @@
...
+# 1. Create a virtual environment and install all dependencies
+uv sync --dev
+
+# 2. Provide local settings (database credentials, etc.)
+cp qatrack/local_settings.example.py qatrack/local_settings.py
+# edit qatrack/local_settings.py as needed
+
+# 3. Apply migrations and load demo data
```
Comment (nsmela) [inline AGENTS.md:138]: The setup sequence is missing two steps and will error. [deploy/dev/local_settings.dev.py](https://github.com/qatrackplus/qatrackplus/blob/develop/deploy/dev/local_settings.dev.py) points sqlite at [db/default.db](https://github.com/qatrackplus/qatrackplus/blob/develop/db/default.db), and settings.py:303 uses DatabaseCache. So mkdir db before migrate, and manage.py createcachetable after it (the developer guide also does collectstatic --noinput before runserver).

File: AGENTS.md [hunk 28]
```diff
@@ -0,0 +1,319 @@
...
+Prefer **LTS releases** for runtime dependencies (Python, Django, Node.js,
+etc.) — clinical deployments need predictable upgrade windows. When proposing
+a new dependency or version bump, note in the PR description whether it's LTS.
+
+### Language
+
+**English (Canada)** is the lightly preferred written language for code
+comments, commit messages, documentation, and user-facing strings. Canadian
+English generally follows British spelling conventions (e.g. *behaviour*,
+*colour*, *programme*) while using common North American technical vocabulary.
```
Comment (nsmela) [inline AGENTS.md:215]: Canadian English keeps program for software in every sense; programme is British and only survives in "programme" as in an event booklet. Since this section is explicitly instructing agents how to spell, that example will actively generate incorrect text. Swap it for something safe like behaviour, colour, licence (noun) — and consider stating outright that program stays program.

File: AGENTS.md (hunk 26 above)
Comment (nsmela) [inline AGENTS.md:157]: no mentions of pre-commit, though the PR checklist gates on uv run pre-commit run --all-files and CONTRIBUTING covers pre-commit install

File: CONTRIBUTING.md
Comment (nsmela) [inline CONTRIBUTING.md:1]: CONTRIBUTING.md should include the standard "contributions are accepted under Apache-2.0" line

File: AGENTS.md [hunk 29]
```diff
@@ -0,0 +1,319 @@
...
+1. **Have read and understood the diff** — not just the AI's summary of it,
+   the actual lines changed.
+2. **Be able to explain the reasoning** behind each non-trivial decision if
+   asked during review.
+3. **Have run the tests locally** and confirmed they pass.
+4. **Take responsibility** for the change's correctness and safety.
+
+Conversely, the maintainers promise:
+
+1. **Embody the spirit of this document** Moderators will not rely on AI as a final review tool.
```
Comment (nsmela) [inline AGENTS.md:38]: Only place moderators is referenced... did we mean Maintainers?

File: AGENTS.md [hunk 30]
```diff
@@ -0,0 +1,319 @@
...
+4. **Take responsibility** for the change's correctness and safety.
+
+Conversely, the maintainers promise:
+
+1. **Embody the spirit of this document** Moderators will not rely on AI as a final review tool. 
+2. **Be honest and transparent** PR review comments will come from a human except where explicitly mentioned. 
+3. **Respect contributors** Contributions are valued, Each maintainer personally commits to manually reviewing each code change, and completing the associated PR checklists.  
+
+Use AI as much as you like to get there — genuinely, that's what it's for.
+Just make sure a human reads what comes out before it ships: Draft PRs are ignored
```
Comment (nsmela) [inline AGENTS.md:43]: belongs in CONTRIBUTING's `Opening a pull request`?

File: AGENTS.md [hunk 31]
```diff
@@ -0,0 +1,319 @@
...
+unless otherwise requested. A Final PR opened autonomously by an AI agent, 
+with no human having read and validated every line, 
+**will be closed without review.** No exceptions.
```
Comment (nsmela) [inline AGENTS.md:46]: against the surrounding promises of patience with newcomers. Consider "closed with a pointer to this policy — reopen once a human has read the diff"

**Review by nsmela — CHANGES_REQUESTED** [review]
- AGENTS.md and CONTRIBUTING.md are already drifting. Beyond the fixtures mismatch: docs build is uv run make docs in one and cd docs && uv run make html in the other (both work — the root Makefile:44 target delegates). The doc-impact table has rows for AGENTS.md ↔ docs/developer/ but nothing pairing CONTRIBUTING.md with AGENTS.md. Either add that row, or have CONTRIBUTING link to AGENTS for the canonical commands instead of repeating them.

File: AGENTS.md (hunk 24 above)
Comment (crcrewso) [reply AGENTS.md:135]: I have a better fix for this so testing will no longer require either local settings files. They're still recommended but they won't be required.

File: AGENTS.md [hunk 32]
```diff
@@ -0,0 +1,337 @@
...
+draft a migration or a test, hunt for bugs, sketch out documentation. The
+maintainer reaches for these tools constantly, and so should you.
+
+Here's the one line we hold firm on: **every pull request needs a human
+owner who has read the diff, understands it, and can defend it.** You
+already know why — this project sits between quality control and patient
+care, and "I don't fully understand what I shipped" isn't a position
+anyone here wants to be in. AI is a great way to get to a first draft
+faster — we can't let it become a substitute for someone standing behind the
+change, that part's still on us.
```
Comment (nsmela) [inline AGENTS.md:25]: opinion: the language used here is pretty informal. Should this be more professional? 

"AI use is permitted as long as **every pull request needs a human owner who has read the diff, understands it, and can defend it.**." 

Not sure if the rest of the wording is needed.

File: CONTRIBUTING.md (hunk 20 above)
Comment (nsmela) [reply CONTRIBUTING.md:120]: resolved: https://github.com/qatrackplus/qatrackplus/pull/840/changes#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R101

**Review by nsmela — CHANGES_REQUESTED** [review]
Comment on language used in AI Agents file. It's an opinion, so feel free to discard.

File: AGENTS.md (hunk 32 above)
Comment (crcrewso) [reply AGENTS.md:25]: I'm not sure, that section is the one section I would like every human to read. I'll think about it. Is it feeling more informal than the README.md? 

I'm converting this back to draft since there have been so many changes since my last complete review that it needs a full readthrough again.

Comment (nsmela) [reply AGENTS.md:25]: understood. ill resolve this and leave it up to you.

**Comment by crcrewso** [comment]
@nsmela — thank you for this review. Fifty-two comments across three passes, and
a good number of them were things I should have caught myself. The branch has
had the full re-read I promised on 2026-09-16 and is squashed to a single
commit, so the old line anchors won't resolve; I've referenced sections instead.

> **Disclosure:** this reply was drafted with AI assistance (Claude), then read,
> checked and posted by me. The AGENTS.md AI policy in this very diff asks for
> exactly that, and it would be a poor look to publish the policy while quietly
> ignoring it. The judgement calls below are mine; the verification of "is this
> actually fixed in the current text" was done with the assistant's help and I
> re-checked it.

## Fixed — the factual corrections

All of these are in the squashed commit.

| Your comment | Where it landed now |
|---|---|
| `only 3.12` | Stack table reads **Python 3.12**; the 3.13 claim is gone |
| `target branch is 'develop'` ×2 | Stack table says ``develop`` (not ``master``), and *Pull requests* repeats it |
| `should be qatrack_core` | Layout now lists ``qatrack_core/`` |
| `no mention of accel_migration_tool` | Listed, with a one-line description |
| `missing admin_media` | Listed |
| `missing cache` | Listed — and you were right about *why*: it's an app providing ``clearcache``, not something ``createcachetable`` generates |
| `whole layout is missing quite a few folders` | Rebuilt from ``ls``. Every tracked app under ``qatrack/`` is listed; only ``static/`` (gitignored — it's the ``collectstatic`` target) and ``__pycache__`` are omitted |
| AGENTS.md — `file does not exist` (docs-impact) | Reworded to the conditional: the workflow **would** post the heuristic, once it exists. No longer claims a file that isn't there |
| CONTRIBUTING — `directory does not exist` | The ``docs/common_issues/`` bullet is removed |
| CONTRIBUTING — `file does not exist` | The docs build no longer references ``requirements/docs.txt``; it's ``uv sync --dev`` then ``cd docs && uv run make html`` |
| `we use ruff now` ×2 | Both replaced. CONTRIBUTING gives ``ruff check`` / ``ruff format`` and points at ``[tool.ruff]`` |
| `need to mention uv run pre-commit --all-files` | In CONTRIBUTING's pre-PR steps **and** AGENTS.md's lint section |
| `no mentions of pre-commit` (AGENTS) | Added, matching the PR checklist that gates on it |
| `pre-commit install should be used` | CONTRIBUTING step 3 of setup |
| `fixtures/demo_data.json does not exist` | Now ``loaddata fixtures/defaults/*/*.json`` — the two files agree |
| setup sequence missing ``mkdir db`` / ``createcachetable`` / ``collectstatic`` | All three added, in that order, to **both** files. This was the best catch of the review: the sequence as written would have failed on the first run for every new contributor |
| `CONTRIBUTING should include the Apache-2.0 line` | Added near the top |
| `moderators… did we mean Maintainers?` | Yes. "moderators" is gone; it's *maintainers* throughout |

## The two I'd flagged as judgement calls — I took your wording

**"reopen once a human has read the diff"** — adopted verbatim. Your phrasing
is better than mine: it turns a closed door into a door with instructions on
it, which is the tone the rest of the section is going for.

**"belongs in CONTRIBUTING's *Opening a pull request*?"** — partly. The policy
stays in AGENTS.md because that's the file an agent reads, and this is the one
section I want read *before* any work starts. But it now links across to
CONTRIBUTING's section explicitly, so a human arriving from either direction
finds the other.

**Canadian English / *programme*** — you were right and it was worse than a
typo: that section instructs agents how to spell, so a wrong example
manufactures wrong text at scale. The examples are now *behaviour*, *colour*,
*licence* (noun), and it says outright that **program stays program**.

## The informality question — thank you for leaving it with me

You asked whether the AI-policy paragraph should read more professionally, and
then resolved it and left the call to me. I've kept the register but tightened
the sentence:

> Here's the one line we hold firm on: **every pull request needs a human owner
> who understands it, and can explain it.** You already know why — this project
> sits between quality control and patient care, and "I don't fully understand
> what I shipped" isn't a position anyone here wants to be in.

My reasoning: this is the one paragraph I want every human to actually read
rather than skim, and the formal register is what makes policy text skimmable.
The four numbered obligations immediately below it carry the precision. If it
still reads as too loose next to the rest of the document, say so and I'll
take another pass — you've earned the veto on this one.

## Still open, deliberately

- **`runtests.sh` doesn't use pytest — should it be removed?** Still out of
  scope here, still the right question. It is now tracked rather than floating.
- **`requirements/dev.txt`** — AGENTS.md carries an explicit TODO instead of a
  wrong description. The real fix is a separate PR.
- **A bug report template** — worth having; not this PR.
- **#801** will not be accepted. I'm leaving it open so that everything in it
  that *isn't* covered here stays visible, including the `docs-impact` workflow
  this file now only speaks about conditionally.
- **Condensing AGENTS.md further** — I agree it's long. Part of that is
  deliberate: as I noted, it is written for agents to read in full, which is a
  different economy from a human document. If AGENTS.md ever becomes something
  humans are expected to read end to end, it should be split rather than
  trimmed.

## One overlap worth knowing about

This PR and #858 both fix the same checklist line in
`.github/pull_request_template.md` — the `makemigrations` step that says
`--check --dry-run`. Django 4.1 made `--check` silent, so a contributor
following that step sees an empty result and concludes there is nothing to
generate; plain `--dry-run` prints what it would create.

The two changes are byte-identical, so they don't conflict and neither blocks
the other — #858 just also fixes the `pre-commit` line next to it. If #858
lands first, that hunk here becomes a no-op. AGENTS.md carries the reasoning
behind the flag choice, which is why the change appears in both places.

**Comment by nsmela** [comment]
> Here's the one line we hold firm on: every pull request needs a human owner
> who understands it, and can explain it. You already know why — this project
> sits between quality control and patient care, and "I don't fully understand
> what I shipped" isn't a position anyone here wants to be in.

My main concern is that this is more glib than professional. I recommend replacing it with:
>Every pull request requires a human owner to understand and explain the included code fully.

Otherwise, approved.

**Review by nsmela — APPROVED** [review]
